src/ComputeSharp.Graphics/Buffers/ConstantBuffer{T}.cs

```diff
@@ -44,31 +44,20 @@ private static int GetPaddedSize()
         public T this[int i] => throw new InvalidExecutionContextException($"{nameof(ConstantBuffer<T>)}<T>[int]");
 
         /// <inheritdoc/>
-        public override void GetData(Span<T> span, int offset, int count)
+        public override unsafe void GetData(Span<T> span, int offset, int count)
         {
             if (IsPaddingPresent)
             {
-                // Create the temporary array
-                byte[] temporaryArray = ArrayPool<byte>.Shared.Rent(count * PaddedElementSizeInBytes);
-                Span<byte> temporarySpan = temporaryArray.AsSpan(0, count * PaddedElementSizeInBytes);
-
                 // Copy the padded data to the temporary array
                 using (MappedResource resource = MapResource())
                 {
-                    MemoryHelper.Copy(resource.Pointer, offset * PaddedElementSizeInBytes, temporarySpan, 0, count * PaddedElementSizeInBytes);
-                }
-
-                ref byte tin = ref temporarySpan.GetPinnableReference();
-                ref T tout = ref span.GetPinnableReference();
-
-                // Copy the padded data to the target span, removing the padding
-                for (int i = 0; i < count; i++)
-                {
-                    ref byte rsource = ref Unsafe.Add(ref tin, i * PaddedElementSizeInBytes);
-                    Unsafe.Add(ref tout, i) = Unsafe.As<byte, T>(ref rsource);
+                    ref var dest = ref MemoryMarshal.GetReference(span);
+                    byte* untypedSrc = (byte*)resource.Pointer;
+                    for (var i = offset; i < count; i++)
```

Comment:
This is not correct, it's not the same behavior as the original code.
The `offset` parameter indicates the offset into the GPU buffer to start getting data from, but that chunk of data is then always copied sequentially from the _start_ of the input `Span<T>`. Here instead you're actually writing items starting from `offset` into the target `Span<T>` too, which is a different behavior.